core/modules/wdb/ChunkResource.cc

```diff
@@ -167,6 +171,11 @@ typedef std::vector<ScTable> ScTableVector;
 
 class Backend {
 public:
+    virtual ~Backend() {
+        if (!_isFake) {
+            _memLockRelease();
```

Comment:
My guess is that release should not be called when it is detected that some other process has this lock. I think `memLockCheck()` calls `exit()` when it detects this situation. Do we have guarantee that Backend's destructor has no chance of being called when exit() happens? I'd probably add some protection in case destructor is called but we detected that lock is not ours.